lib/datadog/tracing/contrib/grpc/datadog_interceptor/client.rb

```diff
@@ -16,33 +17,45 @@ module DatadogInterceptor
           # sending the request to the server.
           class Client < Base
             def trace(keywords)
-              keywords[:metadata] ||= {}
+              formatter = GRPC::Formatting::FullMethodStringFormatter.new(keywords[:method])
 
               options = {
                 span_type: Tracing::Metadata::Ext::HTTP::TYPE_OUTBOUND,
                 service: service_name, # Maintain client-side service name configuration
-                resource: format_resource(keywords[:method])
+                resource: formatter.resource_name
               }
 
               Tracing.trace(Ext::SPAN_CLIENT, **options) do |span, trace|
-                annotate!(trace, span, keywords[:metadata], keywords[:call])
-
-                yield
+                annotate!(trace, span, keywords, formatter)
+
+                begin
+                  yield
+                rescue StandardError => e
+                  code = e.is_a?(::GRPC::BadStatus) ? e.code : ::GRPC::Core::StatusCodes::UNKNOWN
+                  span.set_tag(Contrib::Ext::RPC::GRPC::TAG_STATUS_CODE, code)
+
+                  raise
+                else
+                  span.set_tag(Contrib::Ext::RPC::GRPC::TAG_STATUS_CODE, ::GRPC::Core::StatusCodes::OK)
+                end
               end
             end
 
             private
 
-            def annotate!(trace, span, metadata, call)
-              span.set_tags(metadata)
+            def annotate!(trace, span, keywords, formatter)
+              metadata = keywords[:metadata] || {}
+              call = keywords[:call]
 
-              span.set_tag(Contrib::Ext::RPC::TAG_SYSTEM, Ext::TAG_SYSTEM)
+              span.set_tags(metadata)
```

Comment:
Is this setting all gRPC `metadata` key/value pairs as span tags?
Isn't `metadata` arbitrary user data, meaning it could have PII?
If so, we can't simply tag it like this. Is this a requirement from some internal doc?

Also, the keys will become Datadog tag names, we should make sure they don't conflict our tag namespace, likely by namespacing them to `metadata.METADATA_KEY = METADATA_VALUE`.

Reply:
Hey @marcotc 👋 !
This code was already here before my changes (I just moved the lines around)
I think the concerns that you raise make sense but I think they would be out of scope of this PR. If you wanna discuss this offline please let me know!